io/buffer.ts

```diff
@@ -1,6 +1,14 @@
 import { assert } from "../_util/assert.ts";
 import { copy } from "../bytes/mod.ts";
 
+interface Reader {
```

Comment:
does it make sense to have a `io/types.d.ts` and put the `Reader*`/`Writer*`, etc and then `import type` from there?  It would solve part of https://github.com/denoland/deno/issues/9795.  I wouldn't necessarily refactor all of `io` to use them at the moment, but it would be better than just burying them in `buffer.ts` while all of `io` and other `std` modules should move over to these.

Reply:
That would make more sense and sounds natural. Updated.